tiledb/sm/array_schema/dimension.cc

```diff
@@ -878,12 +878,78 @@ void Dimension::splitting_value(
   assert(unsplittable != nullptr);
 
   auto r_t = (const T*)r.data();
-  T sp = r_t[0] + (r_t[1] - r_t[0]) / 2;
+
+  // Floating-point template specializations for this routine exist below.
+  // All remaining template types are handled in this implementation,
+  // where all template types are integers. We will calculate the split value
+  // with the following expression: `r_t[0] + (r_t[1] - r_t[0]) / 2`. To
+  // prevent overflow in the `(r_t[1] - r_t[0]) / 2` sub-expression, we will
+  // perform this entire sub-expression with a 65-bit bitset.
+  std::bitset<65> r_t0(r_t[0]);
```

Comment:
Shall we make this block a `util` function? Also any way we can check for perf regression, just in case?

Reply:
In the scenario that I've been testing (the multi-fragment read), there isn't a perf regression. This is called infrequently in that scenario and the unit tests, so I'm fairly confident about performance.

As for `util`, I don't think this calculation would be generally useful. If anything, we could ticket some work to build a small utility library for high-level bitset operations. I suspect this won't be the last time we hit overflow and will need to resort to this type of math.

Reply:
Sounds good, no problem.